### PR TITLE
BUG: Fix for wrong cell type when LINE and POLY_LINE present

### PR DESCRIPTION
**Pull request by PranjalSahu**
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Review by dzenanz — APPROVED** [review]

**Review by jhlegarreta — COMMENTED** [review]
Can a test (C++), or test data to an existing test, be added to demonstrate the issue, please?

Can a commit message body be added in order to give an explanation as to why this is required, please?

The above would help hugely. Thanks Pranjal.

**Comment by PranjalSahu** [comment]
@jhlegarreta Yes, I am adding those. It fails for the existing test mesh when writing in binary format.
Also, the test mesh has pixel data that needs to be corrected. For color scalar data in binary mode writing, we convert to uchar which is making the current data value from 0.51 to 0. That data value has to be made uint8.

I assigned the value to cellType variable which permanently modified it for the rest of the loop.

**Comment by jhlegarreta** [comment]
> @jhlegarreta Yes, I am adding those. It fails for the existing test mesh when writing in binary format.
Also, the test mesh has pixel data that needs to be corrected. For color scalar data in binary mode writing, we convert to uchar which is making the current data value from 0.51 to 0. That data value has to be made uint8.

> I assigned the value to cellType variable which permanently modified it for the rest of the loop.

Thanks Pranjal. So maybe that explains why I many of the tests for the mesh IO classes were not working as expected, and I ended up by commenting those lines:
https://github.com/InsightSoftwareConsortium/ITK/pull/3403#issuecomment-1141132531

Would be great if this was the fix that made those work.

File: Modules/IO/MeshBase/include/itkMeshIOBase.h [hunk 1]
```diff
@@ -802,9 +802,13 @@ class ITKIOMeshBase_EXPORT MeshIOBase : public LightProcessObject
         auto numberOfPoints = static_cast<unsigned int>(input[inputIndex++]);
         if (numberOfPoints > 2 && cellType == CellGeometryEnum::LINE_CELL)
         {
-          cellType = CellGeometryEnum::POLYLINE_CELL;
+          auto newCellType = CellGeometryEnum::POLYLINE_CELL;
+          output[outputIndex++] = static_cast<TOutput>(newCellType);
```
Comment (N-Dekker) [inline Modules/IO/MeshBase/include/itkMeshIOBase.h:806]: Please consider merging these two lines to one:

    output[outputIndex++] = static_cast<TOutput>(CellGeometryEnum::POLYLINE_CELL);

It does not seem necessary to introduce an extra non-const variable, "newCellType" here.

**Comment by PranjalSahu** [comment]
@jhlegarreta I am also testing the commented test cases.
Some failures should not happen like a change in the number of points.

**Comment by jhlegarreta** [comment]
On a slightly related note, maybe the Valgrind issues that were uncovered by some of the mentioned tests are related to this too:
https://open.cdash.org/viewDynamicAnalysis.php?buildid=8084453

**Review by thewtex — APPROVED** [review]
:100:

**Comment by PranjalSahu** [comment]
ghostflow-check-master is stuck. I have re-pushed now.

**Comment by dzenanz** [comment]
@bradking @mathstuf do you have some advice regarding ghostflow here? It is obviously misbehaving.

**Comment by mathstuf** [comment]
Do: check

**Pull request merged**